expirationd.lua

```diff
@@ -327,48 +390,202 @@ end
---     * process_while         -- Function to call before checking each tuple;
---                                if it returns false, the task will stop until next full scan.
---     * iterate_with          -- Function which returns an iterator object which provides tuples to check,
---                                considering the start_key, process_while and other options.
---                                There's a default function which can be overriden with this parameter.
---     * on_full_scan_start    -- Function to call before starting a full scan iteration.
---     * on_full_scan_complete -- Function to call after completing a full scan iteration.
---     * on_full_scan_success  -- Function to call after successfully completing a full scan iteration.
---     * on_full_scan_error    -- Function to call after terminating a full scan due to an error.
---     * args                  -- Passed to is_tuple_expired and
---                                process_expired_tuple() as additional context.
---     * full_scan_time        -- Time required for a full index scan (in seconds).
---     * iteration_delay       -- Max sleep time between iterations (in seconds).
---     * full_scan_delay       -- Sleep time between full scans (in seconds).
---     * force                 -- Run task even on replica.
---  }
+--- Run a scheduled task to check and process (expire) tuples in a given space.
+--
+-- How expirationd works in general:
+--
+-- 1. Process min(`space_length`, `tuples_per_iteration`) tuples at once.
+--
+-- 2. Sleep `tuples_per_iteration` × `full_scan_time` / `space_length` (but not
+--    beyond 1 second).
+--
+-- 3. Repeat 1-2 until the whole space will be traversed.
+--
+-- 4. Sleep 1 second.
+--
+-- 5. Repeat 1-4.
+--
+--
+-- @string name
+--     Task name.
+-- @string space_id
+--     Space to look in for expired tuples.
+-- @func is_tuple_expired
+--     Function, must accept tuple and return `true` or `false` (is tuple
+--     expired or not), receives `args` and `tuple` as arguments.
+--
+-- Example of function:
+--
+-- ```
+-- local function is_tuple_expired(args, tuple)
+--     local tuple_expire_time = get_field(tuple, args.field_no)
+--     if type(tuple_expire_time) ~= 'number' then
+--         return true
+--     end
+--     local current_time = fiber.time()
+--
+--     return current_time >= tuple_expire_time
+-- end
+-- ```
+--
+-- @table[opt] options
+--     Table with named options.
+-- @table[opt] options.args
+--     Passed to `is_tuple_expired()` and `process_expired_tuple()` as
+--     an additional context.
+-- @boolean[opt] options.atomic_iteration
+--     False (default) to process each tuple as a single transaction and true
+--     to process tuples from each batch in a single transaction.
+-- @boolean[opt] options.force
+--     expirationd doesn't work on replica by default. With option `force` it
+--     should be possible to force work on replica.
```

Comment:
It deserves a top level note (after this we can say that #14 is resolved). I heard 'why expirationd does not work' so many times...

I would also clarify what exactly 'replica' means in this context.

Reply:
It's good, but I meant a note somewhere at the module level. Maybe as part of usage example. If one don't know that there is a trick around `force` option to make expirationd work on a master with read only upstreams, (s)he will not know where to find the trick. It is the key idea of #14 (the issue appears despite the fact that the force option was mentioned in the readme).

Reply:
Okay, thanks! (I would place it somewhere in the documentation, but README is okay for me too.)